src/transformers/tokenization_utils.py

```diff
@@ -279,87 +410,40 @@ def tokenize(self, text: TextInput, **kwargs) -> List[str]:
             pattern = r"(" + r"|".join(escaped_special_toks) + r")|" + r"(.+?)"
             text = re.sub(pattern, lambda m: m.groups()[0] or m.groups()[1].lower(), text)
 
-        def split_on_token(tok, text):
-            result = []
-            tok_extended = all_special_tokens_extended.get(tok, None)
-            split_text = text.split(tok)
-            full_word = ""
-            for i, sub_text in enumerate(split_text):
-                # AddedToken can control whitespace stripping around them.
-                # We use them for GPT2 and Roberta to have different behavior depending on the special token
-                # Cf. https://github.com/huggingface/transformers/pull/2778
-                # and https://github.com/huggingface/transformers/issues/3788
+        no_split_token = set(self.unique_no_split_tokens)
+        trie = self.trie
+        tokens = trie.split(text)
```

Comment:
Why not directly?
```suggestion
        tokens = self.trie.split(text)
```
It doesn't look like `trie` is used anywhere else.

Reply:
Not sure either why I wrote that in 2 steps :)